pyiso/pjm.py

```diff
@@ -126,3 +146,147 @@ def get_trade(self, latest=False, **kwargs):
+        rename_d = {'price':'lmp',
+                    'pnodeId': 'node_id',
+                    'priceType': 'lmp_type',
+                   }
+        retdf.rename(columns=rename_d, inplace=True)
+        retdf.drop(['publishDate', 'versionNum'], axis=1, inplace=True)
+        retdf['freq'] = self.options['freq']
+        retdf['market'] = self.options['market']
+        retdf['ba_name'] = 'PJM'
+
+        return retdf
+
+    def fetch_dataminer_df(self, endpoint, params):
+        url = self.base_dataminer_url + endpoint
+        response = self.request(url, mode='post', json=params)
+        df = self.parse_dataminer_df(response.json())
+
+        return df
+
+    def handle_options(self, **kwargs):
+        super(PJMClient, self).handle_options(**kwargs)
+
+        if 'market' not in self.options:
+            self.options['market'] = self.MARKET_CHOICES.dam
+
+        if 'freq' not in self.options:
+            self.options['freq'] = self.FREQUENCY_CHOICES.hourly
+
+        if self.options['market'] in (self.MARKET_CHOICES.dam, self.MARKET_CHOICES.hourly):
+            # set correct endpoint for lmp data
+            endpoints = {
+                self.MARKET_CHOICES.dam: '/markets/dayahead/lmp/daily',
+                self.MARKET_CHOICES.hourly: '/markets/realtime/lmp/daily',}
+            self.options['endpoint'] = endpoints[self.options['market']]
+
+        # special handling for five minute lmps
+        elif self.options['market'] == self.MARKET_CHOICES.fivemin:
+            self.options['freq'] = self.FREQUENCY_CHOICES.fivemin
+            # no historical data for 5min lmp
+            self.options['latest'] = True
+
+    def fetch_oasis_data(self):
+        response = self.request(self.oasis_url)
+        dfs = pd.read_html(response.content, header=0, index_col=0, parse_dates=False)
+        df = dfs[1]
+
+        df['node_id'] = df.index
+        df.rename(columns={'5 Minute Weighted Avg. LMP': 'lmp'}, inplace=True)
+        # drop 'Hourly Integrated LMP for Hour Ending XX' and 'Type' columns
+        df.drop([df.columns[2], 'Type'], axis=1, inplace=True)
+
+        # find timestamp
+        soup = BeautifulSoup(response.content)
+        # the datetime is the only bold text on the page, this could break easily
+        ts_elt = soup.find('b')
+
+        # do not pass tzinfos argument to dateutil.parser.parse, it fails arithmetic
+        ts = parse(ts_elt.string, ignoretz=True)
+        ts = pytz.timezone('US/Eastern').localize(ts)
+        ts = ts.astimezone(pytz.utc)
+        df['timestamp'] = ts
+
+        df['freq'] = self.options['freq']
+        df['market'] = self.options['market']
+        df['ba_name'] = 'PJM'
+        df['lmp_type'] = 'TotalLMP'
+        return df
+
+
+    def get_lmp(self, node_id=None, **kwargs):
+        self.handle_options(data='lmp', **kwargs)
+
+        if self.options['market'] == self.MARKET_CHOICES.fivemin:
+            if node_id in self.zonal_aggregate_nodes.keys():
```

Comment:
I am not sure this worth it.  PJM publishes zonal aggregate nodes with prices to the cent, but the rest only to the dime.